src/plugins/azgenerator/TemplateAzureCliCustom.ts

```diff
@@ -90,38 +136,27 @@ function ConstructMethodBodyParameter(model: CodeModelAz, needGeneric: boolean =
         let prefixIndent = "    ";
 
         do {
-            if (model.MethodParameter_IsFlattened) {
-                if (isNullOrUndefined(model.MethodParameter['extensions']?.['cli-poly-as-resource-base-schema'])) {
-                    continue;
-                }
+
+            if (needGeneric && !isNullOrUndefined(genericParameter) && model.MethodParameter_MapsTo == model.Parameter_MapsTo(genericParameter)) {
                 originalParameterStack.push(model.MethodParameter);
                 originalParameterNameStack.push(model.MethodParameter_Name);
```

Comment:
Suppose we push item ins stack here. Then the item should be pop in the same method?

Reply:
Not ready for review.